packages/story-editor/src/app/api/apiProvider.js

```diff
@@ -173,369 +112,141 @@ function APIProvider({ children }) {
-        per_page: perPage,
-        page,
-        _web_stories_envelope: true,
-      });
-      return apiFetch({ path: apiPath }).then(({ headers, body }) => {
-        const totalPages = parseInt(headers['X-WP-TotalPages']);
-        const templates = body.map((template) => {
-          return { ...template['story_data'], templateId: template.id };
-        });
-        return {
-          templates,
-          hasMore: totalPages > page,
-        };
-      });
-    },
-    [customPageTemplates]
+  // @todo Move to wp-story-editor along with meta-boxes.
+  actions.saveMetaBoxes = useCallback(
+    (story, formData) => saveMetaBoxes(story, formData, metaBoxes),
+    [metaBoxes, saveMetaBoxes]
   );
 
-  const addPageTemplate = useCallback(
-    (page) => {
-      return apiFetch({
-        path: `${customPageTemplates}/`,
-        data: {
-          story_data: page,
-          status: 'publish',
-        },
-        method: 'POST',
-      }).then((response) => {
-        return { ...response['story_data'], templateId: response.id };
-      });
-    },
-    [customPageTemplates]
+  // @todo Move to wp-story-editor along with StatusCheck component.
+  actions.getStatusCheck = useCallback(
+    (content) => getStatusCheck(content, statusCheck, encodeMarkup),
+    [statusCheck, encodeMarkup, getStatusCheck]
   );
 
-  const deletePageTemplate = useCallback(
-    (id) => {
-      // `?_method=DELETE` is an alternative solution to override the request method.
-      // See https://developer.wordpress.org/rest-api/using-the-rest-api/global-parameters/#_method-or-x-http-method-override-header
-      return apiFetch({
-        path: addQueryArgs(`${customPageTemplates}${id}/`, {
-          _method: 'DELETE',
-        }),
-        data: { force: true },
-        method: 'POST',
-      });
-    },
-    [customPageTemplates]
+  actions.getCustomPageTemplates = useCallback(
+    (page = 1) => getCustomPageTemplates(page, customPageTemplates),
+    [customPageTemplates, getCustomPageTemplates]
   );
 
-  const state = {
-    actions: {
-      autoSaveById,
-      getStoryById,
-      getDemoStoryById,
-      getStoryLockById,
-      setStoryLockById,
-      deleteStoryLockById,
-      getMedia,
-      getHotlinkInfo,
-      getLinkMetadata,
-      saveStoryById,
-      getAuthors,
-      uploadMedia,
-      updateMedia,
-      deleteMedia,
-      saveMetaBoxes,
-      getStatusCheck,
-      addPageTemplate,
-      getCustomPageTemplates,
-      deletePageTemplate,
-      getPageTemplates,
-      getCurrentUser,
-      updateCurrentUser,
-    },
-  };
+  actions.addPageTemplate = useCallback(
+    (page) => addPageTemplate(page, customPageTemplates),
+    [customPageTemplates, addPageTemplate]
+  );
+
+  actions.deletePageTemplate = useCallback(
+    (id) => deletePageTemplate(id, customPageTemplates),
+    [customPageTemplates, deletePageTemplate]
+  );
+
+  Object.entries(apiCallbacks).forEach(([name, callback]) => {
+    if (!callback) {
```

Comment:
When would that be the case?

Reply:
When some api callbacks are not passed from the configuration. One use case is Playground.

Reply:
Would the playground pass `apiCallbacks: { uploadMedia: null, ... }` or so in that case or simply not pass that at all? If it doesn't pass anything then it's not in `Object.entries` anyway

Reply:
Good catch, the playground can simply not pass at all. Updated code.